thunder/core/transform_common.py

```diff
@@ -415,6 +415,9 @@ def has_tag(bsym: BoundSymbol, tag: prims.OpTags) -> bool:
         if in_tensor in trace_args_set:
             continue
         prod_bsym: BoundSymbol = producer_bsyms[in_tensor]
+        if not prod_bsym.flat_proxy_args:
```

Comment:
would we need to filter for tensors (to exclude numberproxies) when @jjsjann123 's work fully comes to fruition?